### PR TITLE
Reworked /state/non-fungible/* endpoints

### PR DESCRIPTION
**Pull request by KrzysztofLabus-RDX**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.GatewayApi/Validators/StateNonFungibleDetailsRequestValidator.cs [hunk 1]
```diff
@@ -69,16 +69,25 @@
...
 
-        RuleFor(x => x.NonFungibleId)
-            .NotEmpty();
+        RuleFor(x => x.NonFungibleIds)
+            .NotEmpty()
+            .DependentRules(() =>
+            {
+                RuleFor(x => x.NonFungibleIds.Count)
+                    .GreaterThan(0)
+                    .LessThan(endpointOptionsSnapshot.Value.DefaultPageSize);
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.GatewayApi/Validators/StateNonFungibleDetailsRequestValidator.cs:86]: hmm I guess that property was designed to hold default page size if value wasn't provided, not to hold upper limit. 

Guess we should use `MaxPageSize` ? I think it'd also make sense to change default value of MaxPageSize to be higher than default :D currently it's 30 and Default is 100 so if user wouldn't specify anything he'd get 100 items but he can't ask explicitly for more than 30.

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 2]
```diff
@@ -1738,74 +1738,86 @@ components:
...
           properties:
-            address:
-              $ref: "#/components/schemas/EntityAddress"
+            resource_address:
+              $ref: "#/components/schemas/ResourceAddress"
             non_fungible_ids:
               $ref: "#/components/schemas/NonFungibleIdsCollection"
 
-    NonFungibleDataRequest: # TODO rename to NonFungibleIdDataRequest
+    StateNonFungibleDetailsRequest:
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:1769]: If we allow querying by multiple non_fungible_ids shouldn't it be named `StateNonFungible Ids DetailsRequest` ?

**Review by PawelPawelec-RDX — CHANGES_REQUESTED** [review]

File: src/RadixDlt.NetworkGateway.GatewayApi/Validators/StateNonFungibleDetailsRequestValidator.cs (hunk 1 above)
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/Validators/StateNonFungibleDetailsRequestValidator.cs:86]: You're obviously right about both of those issues. Fixed.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs [hunk 3]
```diff
@@ -392,28 +391,34 @@ LIMIT 1
...
+                entityId = entity.Id,
+                nonFungibleIds = nonFungibleIds,
+            },
             cancellationToken: token);
 
-        var data = await _dbContext.Database.GetDbConnection().QueryFirstOrDefaultAsync<NonFungibleIdDataViewModel>(cd);
+        var items = new List<GatewayModel.StateNonFungibleDetailsResponseItem>();
 
-        if (data == null || data.IsDeleted)
+        foreach (var vm in await _dbContext.Database.GetDbConnection().QueryAsync<NonFungibleIdDataViewModel>(cd))
```
[the remark this answers is not in the document]
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs:408]: I disagree, here's my reasoning: `foreach (var vm in await _dbContext.Database.GetDbConnection().QueryAsync<...>(cd))` is already used several times within this very file for slightly more complex operations and my brain's desire for repetition and patterns in general must be fulfilled. :)

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml (hunk 2 above)
Comment (KrzysztofLabus-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:1769]: Not really, not as far as I can tell. Actually here's some quick reading I've found on "nouns of nouns" - https://english.stackexchange.com/questions/255433/how-to-say-a-case-of-multiple-of-multiples-correctly

**Review by PawelPawelec-RDX — APPROVED** [review]

**Pull request merged**